### PR TITLE
fluffychat: 1.10.0 -> 1.11.0, build from source

### PR DESCRIPTION
**Pull request by hacker1024**
###### Description of changes

This builds Fluffychat from source code instead of patching the prebuilt binary.
It builds on the infrastructure added in #201027 to configure the build environment for native dependencies.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by hacker1024** [comment]
Is it really worth maintaining two copies? If it breaks, users can grab the old copy of Nixpkgs with `fetchFromGitHub` until it gets fixed.

**Review by algorithmiker — COMMENTED** [review]
Okay, you convinced me, let's remove fluffychat-bin and hope for the best.

I still think we should store the overrides in build-support but not willing to die on this hill (although flafy shares my opinion).

It will be made irrelevant anyway once https://github.com/NixOS/rfcs/pull/140 is merged

File: pkgs/applications/networking/instant-messengers/fluffychat/default.nix
Comment (genofire) [inline pkgs/applications/networking/instant-messengers/fluffychat/default.nix:9]: v1.11.1 is released:

```suggestion
  version = "1.11.1";
```

File: pkgs/development/compilers/flutter/package-overrides/default.nix
Comment (hacker1024) [reply pkgs/development/compilers/flutter/package-overrides/default.nix:1]: Alright, I'm happy to move it to build-support if that's the majority opinion. Can we merge this first and do that next, though?

File: pkgs/applications/networking/instant-messengers/fluffychat/default.nix
Comment (hacker1024) [reply pkgs/applications/networking/instant-messengers/fluffychat/default.nix]: 1.11.1 is not yet released, but 1.11.0 is. I have upgraded to that.

**Review by algorithmiker — APPROVED** [review]
lgtm if we are going to move to build-support after the merge.

**Review by mkg20001 — APPROVED** [review]
lgtm

**Pull request merged**
